x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/retry/ChatCompletionErrorResponseHandler.java

```diff
@@ -0,0 +1,171 @@
+/*
```

Comment:
Once we transition all the chat completion response handlers to use this class we can remove the methods that are similar within `BaseResponseHandler`.